test/e2e/multikueue/e2e_test.go

```diff
@@ -851,6 +853,8 @@ var _ = ginkgo.Describe("MultiKueue", func() {
 				gomega.Eventually(func(g gomega.Gomega) {
 					updatedCluster := &rayv1.RayCluster{}
 					g.Expect(k8sWorker1Client.Get(ctx, client.ObjectKeyFromObject(raycluster), updatedCluster)).To(gomega.Succeed())
+					// make sure managedBy was set to nil on worker
+					g.Expect(updatedCluster.Spec.ManagedBy).To(gomega.BeNil())
```

Comment:
I would drop this check. I think it is more relevant to check if this creates pods, or runs to completion, than the technicalities of the solution (in e2e tests). 

Such a check belongs conceptually to integration tests imo.